reV/econ/econ.py

```diff
@@ -153,6 +159,228 @@ def __init__(self, points_control, cf_file, cf_year, site_data=None,
+
+            if 'offshore' in self._meta:
+                if self._meta['offshore'].sum() > 1:
+                    w = ('Found offshore sites in econ meta data. '
+                         'This functionality has been deprecated. '
+                         'Please run the reV offshore module to '
+                         'calculate offshore wind lcoe.')
+                    warn(w, OffshoreWindInputWarning)
+                    logger.warning(w)
+
+        elif self._meta is None and self.cf_file is None:
+            self._meta = pd.DataFrame({'gid': self.points_control.sites})
+
+        return self._meta
+
+    @property
+    def time_index(self):
+        """Get the generation resource time index data."""
+        if self._time_index is None:
+            with Outputs(self.cf_file) as cfh:
+                if 'time_index' in cfh.datasets:
+                    self._time_index = cfh.time_index
+
+        return self._time_index
+
+    @staticmethod
+    def _econ_append_pc(pp, cf_file, sites_per_worker=None):
+        """
+        Generate ProjectControls for econ append
+
+        Parameters
+        ----------
+        pp : reV.config.project_points.ProjectPoints
+            ProjectPoints to adjust gids for
+        cf_file : str
+            reV generation capacity factor output file with path.
+        sites_per_worker : int
+            Number of sites to run in series on a worker. None defaults to the
+            resource file chunk size.
+
+        Returns
+        -------
+        pc : reV.config.project_points.PointsControl
+            PointsControl object instance.
+        """
+        try:
+            multi_h5_res, hsds = check_res_file(cf_file)
+            if multi_h5_res:
+                res_cls = MultiFileResource
+                res_kwargs = {}
+            else:
+                res_cls = Resource
+                res_kwargs = {'hsds': hsds}
+
+            with res_cls(cf_file, **res_kwargs) as f:
+                if 'gid' in f.meta:
+                    gid0 = f.meta['gid'].values[0]
+                    gid1 = f.meta['gid'].values[-1]
+                    if (gid0 in pp.df.gid.values
+                            and gid1 in pp.df.gid.values):
+                        # this is the scenario where the meta is a subset
+                        # of the project points such as when econ is being
+                        # run off a single node in append mode with the
+                        # full generation project points.
+                        i0 = pp.index(gid0)
+                        i1 = pp.index(gid1) + 1
+                        pc = PointsControl.split(
+                            i0, i1, pp, sites_per_split=sites_per_worker)
+                    else:
+                        # this is the scenario where the project points
+                        # is a subset of the meta data like for
+                        # test problems.
+                        pc = PointsControl(
```

Comment:
The project points input to this method is extracted from the points control object. Why not just require that points control is the input, then in this case just pass through the PC un-manipulated? 

Also, would this case ever happen? Append mode but the CF gid's are not found in the project points CSV? Should we just raise an exception here? This could be indicative of a mismatch of the project points CSV file and the CF file.